compiler/rustc_parse/src/errors.rs

```diff
@@ -1489,6 +1489,34 @@ pub(crate) struct AttributeOnParamType {
     pub span: Span,
 }
 
+#[derive(Diagnostic)]
+#[diag(parse_attribute_on_type)]
+pub(crate) struct AttributeOnType {
+    #[primary_span]
+    #[label]
+    pub span: Span,
+    #[suggestion(code = "", applicability = "machine-applicable", style = "tool-only")]
+    pub fix_span: Span,
+}
+
+#[derive(Diagnostic)]
+#[diag(parse_attribute_on_generic_type)]
+pub(crate) struct AttributeOnGenericType {
```

Comment:
```suggestion
#[diag(parse_attribute_on_generic_arg)]
pub(crate) struct AttributeOnGenericArg {
```

+ Fluent file update